omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -71,9 +75,76 @@ public long getAllCount()  {
     }
 
     @Override
-    @Test(expected = ResourceDoesNotSupportOperationException.class)
     public void shouldGetAll() throws Exception {
-        super.shouldGetAll();
+        SimpleObject result = deserialize(handle(newGetRequest(getURI())));
+        assertNotNull(result);
+        assertEquals(getAllCount(), Util.getResultsSize(result));
+    }
+    
+    @Test
+    public void shouldUploadReservedIdentifiers() throws Exception {
+        String reservedIdentifiers = 
+                "{\"reservedIdentifiers\": \"1,2,3,4\"}";
+        
+        MockHttpServletRequest getRequest = newGetRequest(getURI() + "/" + SEQUENTIAL_IDENTIFIER_SOURCE_UUID);
+        getRequest.addParameter("v", "custom:(reservedIdentifiers)");
+        SimpleObject initialIdentifiers = deserialize(handle(getRequest));
+        assertEquals("[]", initialIdentifiers.get("reservedIdentifiers").toString());
+        
+        MockHttpServletRequest postRequest = newPostRequest(getURI() + "/" + SEQUENTIAL_IDENTIFIER_SOURCE_UUID, reservedIdentifiers);
+        SimpleObject identifiersAfterUpload = deserialize(handle(postRequest));
+        assertEquals("[1, 2, 3, 4]", identifiersAfterUpload.get("reservedIdentifiers").toString());
+    }
+    
+    @Test
+    public void shouldGenerateIdentifiers() throws Exception {
+        String generateIdentifiers = 
+                "{\"generateIdentifiers\": \"true\"," +
+                "\"comment\": \"new remote identifiers\"," + 
+                "\"numberToGenerate\": \"5\"}";
+        MockHttpServletRequest request = newPostRequest(getURI() + "/" + SEQUENTIAL_IDENTIFIER_SOURCE_UUID, generateIdentifiers);
+        SimpleObject result = deserialize(handle(request));
+        assertNotNull(result);
+        assertEquals("[G-0, H-8, I-5, J-3, K-1]", result.get("identifiers"));
+    }
+    
+    @Test
+    public void shouldUploadIdentifiersFromSource() throws Exception {
+        String uploadIdentifiers = 
```

Comment:
Did you intentionally decide to use a name different from that of the actual batch size property?

Reply:
Yes I did. I thought it would be clearer on what that section of code does. We are  not just saving but uploading a batch of identifiers.